sotodlib/hwp/g3thwp.py

```diff
@@ -322,145 +424,73 @@ def analyze(self, data, ratio=0.25, mod2pi=True, fast=True):
-            rising_edge = rising_edge_2
-            counter = data['counter_2']
-            counter_index = data['counter_index_2']
-            quad_time = data['quad_2'][0]
-            quad = self._quad_form(data['quad_2'][1])
-
-        fast_time = np.array([])
-        angle = np.array([])
-        hwp_rate = np.array([])
-        if len(counter) > 0 and len(irig_time) > 0:
-            fast_time, angle = self._hwp_angle_calculator(
-                counter, counter_idx, irig_time, rising_edge, quad_time, quad, ratio, mod2pi, fast)
-            if len(fast_time) == 0:
-                return {}
 
-            # hwp speed calc. (approximate using ref)
-            hwp_rate_ref = 1 / np.diff(fast_time[self._ref_indexes])
-            hwp_rate = [hwp_rate_ref[0] for i in range(self._ref_indexes[0])]
-            for n in range(len(np.diff(self._ref_indexes))):
-                hwp_rate += [hwp_rate_ref[n]
-                             for r in range(np.diff(self._ref_indexes)[n])]
-            hwp_rate += [hwp_rate_ref[-1] for i in range(len(fast_time) -
-                                                         self._ref_indexes[-1])]
-
-        # Slow block
-        # - Time definition -
-        # if fast_time exists: slow_time = fast_time
-        # elif: irig_time exists but no fast_time, slow_time = irig_time
-        # else: slow_time is per 10 sec array
-        if len(fast_time) != 0 and len(irig_time) != 0:
-            fast_irig_time = fast_time
-            locked = np.ones(len(fast_time), dtype=bool)
-            locked[np.where(hwp_rate == 0)] = False
-            stable = np.ones(len(fast_time), dtype=bool)
-
-            irig_only_time = irig_time[np.where(
-                (irig_time < fast_time[0]) | (irig_time > fast_time[-1]))]
-            irig_only_locked = np.zeros(len(irig_only_time), dtype=bool)
-            irig_only_hwp_rate = np.zeros(len(irig_only_time), dtype=float)
-            fast_irig_time = np.append(irig_only_time, fast_time).flatten()
-            fast_irig_idx = np.argsort(fast_irig_time)
-            fast_irig_time = fast_irig_time[fast_irig_idx]
-            locked = (np.append(irig_only_locked, locked).flatten())[
-                fast_irig_idx]
-            hwp_rate = (np.append(irig_only_hwp_rate, hwp_rate).flatten())[
-                fast_irig_idx]
-            stable = np.ones(len(fast_irig_time), dtype=bool)
-        elif len(fast_time) == 0 and len(irig_time) != 0:
-            fast_irig_time = irig_time
-            locked = np.zeros(len(fast_irig_time), dtype=bool)
-            stable = np.ones(len(fast_irig_time), dtype=bool)
-            hwp_rate = np.zeros(len(fast_irig_time), dtype=float)
-        else:
-            fast_irig_time = []
-            locked = []
-            stable = []
-            hwp_rate = []
+        # hwp angle calc.
+        if ratio is not None:
+            logger.info(f"Overwriting reference slit threshold by {ratio}.")
+            self._ref_range = ratio
```

Comment:
comment about small bug I found.

Previously the `ratio` parameter of `analyze` was not doing anything.
I modified to be able to overwrite the ratio. 
This option is useful in debug process.